### ISSUE TITLE
Signature matching of functional types

### ISSUE THREAD

**Issue by objeck** [objeck/objeck-lang#163]
The following code would not compile

```
app->SignalConnect("activate", Activate(GObject, Base) ~ Nil, "Hi Ya!");
```

to

```
function : Activate(object : GObject, data : Base) ~ Nil { ... }
```

The parsing of signature types needed to be corrected for nested instances.

**Comment by objeck** [comment]
Fixed in 3cf96770119934f4bb24db58ab901e0140b35fbb

**Comment by objeck** [comment]
Deployed in e53cddd
